sample/Pizzeria.Auth/IAuthServerClient.cs

```diff
@@ -0,0 +1,110 @@
+namespace Pizzeria.Auth;
+
+/// <summary>
+/// Interface for authentication server operations
+/// </summary>
+public interface IAuthServerClient
+{
+    /// <summary>
+    /// Gets an admin access token for the authentication server
+    /// </summary>
+    /// <param name="cancellationToken">Cancellation token</param>
+    /// <returns>Admin access token</returns>
+    Task<string> GetAdminAccessTokenAsync(CancellationToken cancellationToken);
+
+    /// <summary>
+    /// Sets the bearer token for subsequent requests
+    /// </summary>
+    /// <param name="token">Bearer token</param>
+    void SetBearerToken(string token);
```

Comment:
I don't think these are needed.

Instead, use a delegate handler to get an admin token and set it as the authorization header.

Use memory cache to hold the token until just before its token expiry so that it does not need to be read for each call this HTTP client.